src/source/Ice/IceAgent.c

```diff
@@ -1304,10 +1305,24 @@ STATUS incomingDataHandler(UINT64 customData, PSocketConnection pSocketConnectio
         locked = FALSE;
 
         pIceAgent->iceAgentCallbacks.inboundPacketFn(pIceAgent->customData, pBuffer, bufferLen);
-        CHK(FALSE, retStatus);
-    }
+    } else {
+        CHK_STATUS(handleStunPacket(pIceAgent, pBuffer, bufferLen, pSocketConnection, pSrc, pDest,
+                &pStunResponse, &useTurn));
+
+        if (pStunResponse != NULL) {
```

Comment:
this way is definitely better than creating threads.

Reply:
Thanks. I do think that we will need a separate service thread that services all of the outgoing packets with some buffering.